crates/task-impls/src/transactions.rs

```diff
@@ -240,11 +240,36 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> TransactionTaskState<TYPES,
             .as_ref()
             .is_some_and(|cert| cert.upgrading_in(block_view))
         {
-            if let Ok(bundles) = self
-                .auction_results_provider
-                .fetch_bundles(block_view)
-                .await
+            let start = Instant::now();
+
+            if let Ok(Ok(urls)) = async_timeout(
+                self.builder_timeout,
+                self.auction_results_provider
+                    .fetch_auction_result(block_view),
+            )
+            .await
             {
+                let mut futures = Vec::new();
+
+                for url in urls.urls() {
+                    futures.push(async_timeout(
+                        self.builder_timeout.saturating_sub(start.elapsed()),
+                        async {
+                            let client = crate::builder::v0_3::BuilderClient::new(url);
+                            client.bundle(*block_view).await
+                        },
+                    ));
+                }
+
+                let mut bundles = Vec::new();
+
+                for bundle in join_all(futures).await {
+                    match bundle {
+                        Ok(Ok(b)) => bundles.push(b),
+                        _ => continue,
+                    }
+                }
+
```

Comment:
This section looks good to me

Reply:
Looks fine to me too, though I think it would be nice to have this logic be a separate function. We can make an issue for that/do it later though.

Reply:
Yeah, we can do it when implementing this part of the spec:
> Query all builders in parallel, have a timeout that is constrained by a max system timeout of 1 second, and a minimum set by 2/3 of the builders, then add 10% additional time for the remaining 1/3 to resolve.

Reply:
Although I'm not opposed to factoring it out now, won't take long.